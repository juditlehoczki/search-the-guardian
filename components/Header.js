import React from "react";
import { View, Text, TouchableOpacity, Image } from "react-native";

import s from "../styles.js";

const Header = (props) => {
  const { refreshMainPage } = props;
  return (
    <View>
      <Text style={s.search}>search</Text>
      <TouchableOpacity
        onPress={() => {
          refreshMainPage();
        }}
      >
        <Image style={s.logo} source={require("../img/theguardian-logo.png")} />
      </TouchableOpacity>
    </View>
  );
};

export default Header;
