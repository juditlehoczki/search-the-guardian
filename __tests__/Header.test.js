import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';

import Header from '../components/Header.js';

let fakeRefreshMainPage;
let props;

describe('Header Component', () => {
  beforeEach(() => {
    fakeRefreshMainPage = jest.fn();
    props = {
      refreshMainPage: fakeRefreshMainPage,
    };
  });

  test('Snapshot - it renders correctly', () => {
    const snapshot = renderer.create(<Header />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('Image: on press it calls refreshMainPage function', () => {
    const fakeHeader = renderer.create(<Header {...props} />);
    const image = fakeHeader.root.findByType(TouchableOpacity);
    image.props.onPress();
    expect(fakeRefreshMainPage).toHaveBeenCalled();
  });
});
