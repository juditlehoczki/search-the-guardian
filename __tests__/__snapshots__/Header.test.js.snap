// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot - it renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#54C1B7",
        "fontSize": 20,
        "fontWeight": "bold",
        "letterSpacing": 10,
        "marginBottom": -27,
        "marginTop": 80,
        "textAlign": "center",
      }
    }
  >
    search
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 100,
          "paddingVertical": 10,
          "resizeMode": "contain",
          "width": "95%",
        }
      }
    />
  </View>
</View>
`;
